lib/crypto/sig_verifier.go

```diff
@@ -1,26 +1,28 @@
 // Copyright 2021 ChainSafe Systems (ON)
 // SPDX-License-Identifier: LGPL-3.0-only
 
-package runtime
+package crypto
 
 import (
-	"fmt"
+	"errors"
 	"sync"
 	"time"
 
 	"github.com/ChainSafe/gossamer/internal/log"
-	"github.com/ChainSafe/gossamer/lib/crypto"
-	"github.com/ChainSafe/gossamer/lib/crypto/ed25519"
-	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
-	"github.com/ethereum/go-ethereum/crypto/secp256k1"
 )
 
+// ErrSignatureVerificationFailed usaged when signature verification fails
+var ErrSignatureVerificationFailed = errors.New("failed to verify signature")
+
+// SigVerifyFunc verify an signature given a pubkey and msg
+type SigVerifyFunc func(pubkey, sig, msg []byte) (ok bool, err error)
+
 // Signature ...
 type Signature struct {
```

Comment:
A bit out of scope, but this is certainly not a `Signature`, maybe rename it to i.e. `Signing` or `SigningVerifier`? :thinking: